src/test/ui/issues/issue-58856-2.stderr

```diff
@@ -0,0 +1,30 @@
+error: expected one of `)` or `,`, found `->`
+  --> $DIR/issue-58856-2.rs:6:26
+   |
+LL |     fn how_are_you(&self -> Empty {
+   |                   -     -^^
+   |                   |     |
+   |                   |     help: `)` may belong here
+   |                   unclosed delimiter
+
+error: expected one of `async`, `const`, `crate`, `default`, `existential`, `extern`, `fn`, `pub`, `type`, `unsafe`, or `}`, found `)`
+  --> $DIR/issue-58856-2.rs:11:1
+   |
+LL |     }
+   |      - expected one of 11 possible tokens here
+LL | }
+   | ^ unexpected token
```

Comment:
Finally identified what is causing these: when reaching the end of the current `TokenTree`, `self.bump()` will advance to a close token corresponding to the opening delimiter. There are two options: 1) attempt to rewrite the `TokenTree` representation that the `Parser` has once we suggest a close delim (high potential for bad lurking bugs), or 2) catch this specific case in `expect_one_of`, but that might be too magical.